### PR TITLE
Add applicant screen notification bar

### PR DESCRIPTION
**Pull request by rayepps**
**Release Type:** *Non-Breaking Feature*

Fixes https://x-team-internal.atlassian.net/browse/XP-3376

## Description

Adds two new components (and stories for each one - also with knobs):
- `NotificationBar`
- `ApplicantScreenNotification` (which composes `NotificationBar`

then adds the `ApplicantScreenNotification` to the `ApplicantScreen`.

## Checklist
- [x] set yourself as an assignee
- [x] set appropriate labels for a PR (`In Review` or `In Progress` depending on its status)
- [x] move respective JIRA issue to the `IN REVIEW` column
- [x] make sure your code lints (`npm run lint`)
- [x] Flow typechecks passed (`npm run flow`)
- [x] Snapshots tests passed (`npm run jest`)
- [x] check cleanup tasks (https://github.com/x-team/xp/labels/cleanup) and take a suitable small one (if exists) in a related area of the current changes
- [x] component's documentation (`.stories.js` file) is changed or added accordingly to reflect any new or updated use cases or variants usage
- [x] if any snapshots have been changed, verify that component still works and looks as expected and update the changed snapshot
- [x] **manually tested the app** by running it in several different browsers (Firefox, Chrome, Opera, Safari, MS IE/Edge, etc.) and checked nothing is broken and operates as expected!

## Steps to Test or Reproduce

- On storybook (https://deploy-preview-465--xpui-develop.netlify.com/?path=/story/screens-and-layouts-applicantscreen--basic-usage)
- observe the notification bar is present
- open the knobs tab
- toggle the know and view the notification go away and come back

## Impacted Areas in Application

`ApplicantScreen` component

## Screenshots

**ApplicantScreen:**
image

**ApplicantScreenNotification:**
image

**NotificationBar:**
image

### CODE REVIEW

**Comment by nicksp** [comment]
@rayepps seems like for 2 new components are gonna be used specifically inside existing `ApplicantScreen`, so probably we won't need to export them. cc @bernardodiasc

File: src/components/ui/ApplicantScreen.stories.js [hunk 1]
```diff
@@ -13,7 +14,22 @@ const Body = ({ children }) => (
   </div>
 )
 
-storiesOf('Screens and Layouts|ApplicantScreen', module)
+const defaultStories = storiesOf('Screens and Layouts|ApplicantScreen', module)
+
+defaultStories.addDecorator(withKnobs)
```
Comment (bernardodiasc) [inline src/components/ui/ApplicantScreen.stories.js]: I believe this decorator is not needed. Please take a look at how other stories uses `withKnobs`.

File: src/components/ui/ApplicantScreenNotification.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import theme from '../../styles/theme'
+import typo from '../../styles/typo'
+
+const cmz = require('cmz')
+
+const cx = {
+  wrapper: cmz(`
+    margin-left: 40px
+  `),
+  text: cmz(
```
Comment (bernardodiasc) [inline src/components/ui/ApplicantScreenNotification.js]: Let's use a space in between classes. This is to make it consistent with overall usage, the benefit is more visible when have lots of classes but we might want to keep things consistent anyway. :)

File: src/components/ui/ApplicantScreenNotification.stories.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import React from 'react'
+import { storiesOf } from '@storybook/react'
+import { withKnobs, boolean } from '@storybook/addon-knobs'
+
+import ApplicantScreenNotification from './ApplicantScreenNotification'
+
+const defaultStories = storiesOf('UI Components|ApplicantScreenNotification', module)
+
+defaultStories.addDecorator(withKnobs)
```
Comment (bernardodiasc) [inline src/components/ui/ApplicantScreenNotification.stories.js]: I believe this decorator is not needed. Please take a look at how other stories uses `withKnobs`.

File: src/components/ui/NotificationBar.stories.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+import React from 'react'
+import { storiesOf } from '@storybook/react'
+import { withKnobs, boolean } from '@storybook/addon-knobs'
+
+import NotificationBar from './NotificationBar'
+
+const defaultStories = storiesOf('UI Components|NotificationBar', module)
+
+defaultStories.addDecorator(withKnobs)
```
Comment (bernardodiasc) [inline src/components/ui/NotificationBar.stories.js]: I believe this decorator is not needed. Please take a look at how other stories uses `withKnobs`.

File: src/components/ui/SvgIcon.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ import theme from '../../styles/theme'
 const cmz = require('cmz')
 
 export type Icon = 'cog' | 'head' | 'webcam' | 'message' | 'terminal' | 'diamond' | 'talking' | 'trophy' | 'plus' | 'minus' | 'calendar' | 'trashcan' | 'trashcanAlt' | 'x' | 'add' | 'triangleup' | 'triangledown' | 'hamburger' | 'magnifier' | 'edit' | 'check' | 'paperplane' | 'archive' | 'list' | 'grid' | 'dock' | 'github' | 'linkedin' | 'stackoverflow' | 'filter' | 'show' | 'play' | 'pause' | 'redirect' | 'link' | 'basecrm' | 'plusquare' | 'time' | 'spin' | 'filters' | 'arrowleft' | 'copy'
-export type Color = 'default' | 'inverted' | 'monochrome' | 'grayscale' | 'text' | 'mutedgray' | 'grayscarpaflow' | 'frenchGrayDarker' | 'green'
+export type Color = 'default' | 'inverted' | 'monochrome' | 'grayscale' | 'text' | 'mutedgray' | 'grayscarpaflow' | 'frenchGrayDarker' | 'green' | 'white'
```
Comment (bernardodiasc) [inline src/components/ui/SvgIcon.js]: I know the name is confusing, this is very old stuff, probably we should review that. But anyway, I'd suggest to use the `inverted` here and not add a new redundant one. The `inverted` is actually white.

File: src/index.js [hunk 6]
```diff
@@ -8,6 +8,7 @@ import ApplicantBadge from './components/ui/ApplicantBadge'
 import ApplicantGrid from './components/ui/ApplicantGrid'
 import ApplicantProfileContentDisplay from './components/ui/ApplicantProfileContentDisplay'
 import ApplicantScreen from './components/ui/ApplicantScreen'
+import ApplicantScreenNotification from './components/ui/ApplicantScreenNotification'
```
Comment (bernardodiasc) [inline src/index.js]: We probably don't need to add those new components here because they will not be used in XP.

Comment (nicksp) [reply src/index.js]: Yep, that's a point from my review as well.

**Comment by nicksp** [comment]
> One thing is not much clear to me is why you created the NotificationBar? Isn't something that could live inside ApplicantScreenNotification? I'm asking because I could not see a clear distinction between the two. Maybe I didn't get the reason of this decision, if you can explain I would appreciate! Regardless of that, some styles that should belong to NotificationBar exists in ApplicantScreenNotification, for instance the text styles.

That's a good point to consider moving `NotificationBar` into `ApplicantScreenNotification` unless there's a reason to make it separate 👍 

> Another thing I notice is the wrapper margin, I believe it should follow same principle of JobsPageLayout or HeaderBar, that's accordingly design. Right now it's hardcoded but probably something we'll standardise in a grid system soon. Basically the wrapper should have max width 1100px.

This seems to overlap with what I've [shared before](https://github.com/x-team/xp-ui/pull/465#pullrequestreview-306497902), so seems like we're on the same page here and could be a good candidate to improve 😄 

cc @rayepps

File: src/components/ui/ApplicantScreen.stories.js (hunk 1 above)
Comment (rayepps) [reply src/components/ui/ApplicantScreen.stories.js]: Screen Shot 2019-10-24 at 10 42 44 PM

Hey @bernardodiasc could you give an example? I can't find any other stories that use knobs. Also, I found this in the add-ons GitHub documentation.

File: src/components/ui/SvgIcon.js (hunk 5 above)
Comment (rayepps) [reply src/components/ui/SvgIcon.js]: done 👍 [a9c31c4](https://github.com/x-team/xp-ui/pull/465/commits/a9c31c48b12618f1179ad6341f19ae09866fb937)

File: src/index.js (hunk 6 above)
Comment (rayepps) [reply src/index.js]: done 👍 [a9c31c4](https://github.com/x-team/xp-ui/pull/465/commits/a9c31c48b12618f1179ad6341f19ae09866fb937)

File: src/components/ui/ApplicantScreen.stories.js (hunk 1 above)
Comment (nicksp) [reply src/components/ui/ApplicantScreen.stories.js]: We have several examples of where we use knobs. Just search for `with knobs` inside `xp-ui`: `RichTextEditor.stories.js`, `Email.stories.js` are some of examples

**Comment by bernardodiasc** [comment]
@nicksp I made the adjustments we discussed in https://github.com/x-team/xp-ui/pull/465/commits/75d075c6cdfc2f19052c4c68bc39c2e16efdcda4

Please make sure to run `npm install` to get the `react-transition-group`. 

> In the GIFs below, the recording of the animation is kinda laggy, please disregard that issue, try it on live showcase instead

- https://deploy-preview-465--xpui-develop.netlify.com/?path=/story/screens-and-layouts-applicantscreen-states--with-notification



- https://deploy-preview-465--xpui-develop.netlify.com/?path=/story/ui-components-applicantscreennotification--basic-usage

**Comment by nicksp** [comment]
@bernardodiasc Would you mind rebasing to resolve merge conflict here?

**Comment by bernardodiasc** [comment]
@nicksp I rebased and fixed up some commits, so now this PR only contain 2 (first and second implementations)

**Pull request merged**